### PR TITLE
Working Project, adding some typescript

### PR DESCRIPTION
**Pull request by unachoza**

### CODE REVIEW

File: src/API/useFetch.tsx [hunk 1]
```diff
@@ -1,20 +1,27 @@
 //@ts-nocheck
-import React, { useEffect, useState } from "react";
+import { useEffect, useState } from "react";
 import axios from "axios";
 const GITHUB_BASE_URL = 'https://api.github.com';
 
+export interface repositoryCommitMessage  {
+  date: {}
```
Comment (AndraLally) [inline src/API/useFetch.tsx]: when using TY you should type the keys inside the object and thier values. 

```js
export interface repositoryCommitMessage  {
  date: {
  value: string;
  otherValue: string;
  }
}           
```

File: src/API/useFetch.tsx [hunk 2]
```diff
@@ -1,20 +1,27 @@
...
+  commitMessage: "string"
+  url: "string",
+  username: "string"
```
Comment (AndraLally) [inline src/API/useFetch.tsx]: This snytax looks off to me. Should it not just be the word `string` without the `""`?  Right now you are just setting the username type you the string `"string"` if that makes sense.

File: src/API/useFetch.tsx [hunk 3]
```diff
@@ -1,20 +1,27 @@
 //@ts-nocheck
-import React, { useEffect, useState } from "react";
+import { useEffect, useState } from "react";
 import axios from "axios";
 const GITHUB_BASE_URL = 'https://api.github.com';
 
+export interface repositoryCommitMessage  {
```
Comment (AndraLally) [inline src/API/useFetch.tsx]: Interfaces are normally capitalized.

File: src/API/useFetch.tsx [hunk 4]
```diff
@@ -1,20 +1,27 @@
...
+  date: {}
+  commitMessage: "string"
+  url: "string",
+  username: "string"
+} 
+interface repositoryCommitMessages extends Array<repositoryCommitMessage>{}
 
 export const repositoryFetch = async( queryText: string)  => {
```
Comment (AndraLally) [inline src/API/useFetch.tsx]: If you want you can type what this func returns.

File: src/API/useFetch.tsx [hunk 5]
```diff
@@ -1,20 +1,27 @@
...
   const data = await axios.get(`${GITHUB_BASE_URL}/search/repositories?q=${encodeURIComponent(queryText)}`);
```
Comment (AndraLally) [inline src/API/useFetch.tsx]: Let's do some fancy sytax here using less code. 

```js
  const {data}  = await axios.get(`${GITHUB_BASE_URL}/search/repositories?q=${encodeURIComponent(queryText)}`);
  const repositoryChoices = shapeData(data.items, normalizeRepoData)
```

File: src/API/useFetch.tsx [hunk 6]
```diff
@@ -1,20 +1,27 @@
...
   const items = data.data.items
   const repositoryChoices = shapeData(items, normalizeRepoData)
```
Comment (AndraLally) [inline src/API/useFetch.tsx]: `normalizeData` is the term you are looking for not `shape`

File: src/API/useFetch.tsx [hunk 7]
```diff
@@ -1,20 +1,27 @@
...
-  console.log({ repositoryChoices })
   return repositoryChoices
 };
 
 
-export const useFetch = (repositoryOwner: string, repositoryName: string) => {
-  const [data, setData] = useState([])
+
+export const useFetch = (repositoryOwner: string, repositoryName: string, fire: boolean) => {
+  const [data, setData ] = useState([])
```
Comment (AndraLally) [inline src/API/useFetch.tsx]: This is fancy, good job. You can make it fancier this way.  Also, remove the comment at the bottom of the page.
https://kentcdodds.com/blog/stop-using-isloading-booleans

File: src/App.tsx [hunk 8]
```diff
@@ -1,67 +1,31 @@
...
-  const [loadingState, setLoadingState] = useState<LoadingState>(INITIALSTATE);
-  const [repositoryChoices, setRepositoryChoices] = useState([]);
-  const [repositoryQuery, setRepositoryQuery] = useState("");
-  const [selectedRepo, setSelectedRepo] = useState({})
   const [repositoryQueryOwner, setRepositoryQueryOwner] = useState("")
   const [repositoryQueryTitle, setRepositoryQueryTitle] = useState("")
-  const [commits, setCommits] = useState([]);
-
-const {data, error, loading} = useFetch(repositoryQueryOwner,repositoryQueryTitle )
+  const [fire, setFire] = useState(false);
```
Comment (AndraLally) [inline src/App.tsx]: what does fire mean? you can call these anything you want.

File: src/App.tsx [hunk 9]
```diff
@@ -1,67 +1,31 @@
...
-        {/* {repositoryChoices.length > 0 ? (
-          <> */}
-        <CardList loading={loadingState} data={data} title={repositoryQueryTitle}/>
-            <Button text="Load More" onClick={() => fetchCommits(repositoryQueryOwner, repositoryQueryTitle)} />
-          {/* </>
-        ) : (
-            null
-          // <Loading />
-        )} */}
+        <CardList data={data} title={repositoryQueryTitle}/>
```
Comment (AndraLally) [inline src/App.tsx]:
```js
cards={data}
```

File: src/Components/Button/Button.tsx [hunk 10]
```diff
@@ -1,7 +1,12 @@
...
 import React from "react";
+interface IProps_Button {
+  text: string;
+  onClick: React.MouseEventHandler<HTMLButtonElement>;
+}
 
-const Button = ({ text, onClick }) => {
+
+const Button = ({ text, onClick }: IProps_Button): JSX.Element  => {
   return <div className="button" onClick={onClick}>{text}</div>;
```
Comment (AndraLally) [inline src/Components/Button/Button.tsx]: Gurl tell me what is very wrong here?

File: src/Components/Card/Card.tsx [hunk 11]
```diff
@@ -1,21 +1,21 @@
...
-      <div className="date">{commit[1].date.day}</div>
-      <div className="time">{commit[1].date.time}</div>
-      </div>
-    <div >{commit[1].commitMessage}</div>
-      <div className="author">{commit[1].username}</div>
-    </a>
-  </>
-  )
+const Card = (props:any ): JSX.Element => {
+  const {commit} = props;
```
Comment (AndraLally) [inline src/Components/Card/Card.tsx:4]: Only pass in `commit[1]` as commit.

File: src/Components/CardList/CardList.tsx [hunk 12]
```diff
@@ -1,25 +1,25 @@
...
+      {data.length === 1 ? (
         <Card title={data[0]} />
       ) : (
         <ul>
           {Object.entries(data).map((commit, i) =>
             data.length === i + 1 ? (
-              <div key={i}>
-                {commit.title}
-              </div>
+              <div key={i}>{commit.title}</div>
```
Comment (AndraLally) [inline src/Components/CardList/CardList.tsx]: Don't use the index this is what React uses by default. Use something off the commit.

Comment (AndraLally) [reply src/Components/CardList/CardList.tsx]: Im confused here. Why is data an object I feel like you need an array?

File: src/Components/CardList/CardList.tsx [hunk 13]
```diff
@@ -1,25 +1,25 @@
...
+}
+const CardList = (props: CardListProps) => {
+  const {data, title} = props;
   return (
     <div className="commit__container">
-      <div className="title">Showing Results For <strong>{title} </strong></div>
-      {!data.length ? (
-        <div className="no-results">
-          Oh dear!<br></br> Your search returned no results.
+      {data.length ? (
```
Comment (AndraLally) [inline src/Components/CardList/CardList.tsx]: all this logic can be cleaned up a lot. Give it a try and let me look at it again. You have too much syntax here.

File: src/Components/CardList/CardList.tsx [hunk 14]
```diff
@@ -28,16 +28,12 @@ const CardList = ({ data , title}) => {
...
     </div>
   );
 };
 
 export default CardList;
 
-
-
-
 // const Card = () => {
```
Comment (AndraLally) [inline src/Components/CardList/CardList.tsx]: delete call commnets.

File: src/API/useFetch.tsx (hunk 2 above)
Comment (unachoza) [reply src/API/useFetch.tsx]: you're right!

File: src/Components/Button/Button.tsx (hunk 10 above)
Comment (unachoza) [reply src/Components/Button/Button.tsx]: @AndraLally maybe the name "onClick"

File: src/API/useFetch.tsx (hunk 5 above)
Comment (unachoza) [reply src/API/useFetch.tsx]: no longer using this function

**Pull request merged**